app/controllers/api/chat_channels_controller.rb

```diff
@@ -13,13 +13,23 @@ def update
     end
 
     editable_params = params.permit(*CHAT_CHANNEL_EDITABLE_PARAMS)
+    editable_params.delete(:auto_join_users) if !chat_channel.category_channel?
     editable_params.each do |k, v|
       editable_params[k] = nil if editable_params[k].blank?
     end
+
+    if ActiveRecord::Type::Boolean.new.deserialize(editable_params[:auto_join_users])
```

Comment:
How about we just cancel the previous jobs ?

Reply:
Updating the `auto_join_users` attribute will make all batches to return early, so there's no need to cancel.

Also, we don't queue the batch jobs directly, but queue a different job that does it for us. It means we'll introduce a race condition when we attempt to cancel them. I'm ok with adjusting the rate limiter if it's too restrictive